### PR TITLE
[exporter/datadog] Respect confighttp in trace agent

### PR DESCRIPTION
**Pull request by songy23**

### CODE REVIEW

File: exporter/datadogexporter/traces_exporter.go [hunk 1]
```diff
@@ -33,6 +34,13 @@ import (
 	"github.com/open-telemetry/opentelemetry-collector-contrib/exporter/datadogexporter/internal/scrub"
 )
 
+var traceCustomHTTPFeatureGate = featuregate.GlobalRegistry().MustRegister(
+	"exporter.datadogexporter.TraceAgentUseCustomHTTPClient",
```
Comment (mx-psi) [inline exporter/datadogexporter/traces_exporter.go:38]: I don't think 'trace agent' is a meaningful concept for end-users of the Datadog exporter: it's an implementation detail. Is there any way to rephrase this to make it more meaningful to end-users?

File: exporter/datadogexporter/traces_exporter.go [hunk 2]
```diff
@@ -212,6 +220,12 @@ func newTraceAgentConfig(ctx context.Context, params exporter.Settings, cfg *Con
 	acfg.PeerTagsAggregation = cfg.Traces.PeerTagsAggregation
 	acfg.PeerTags = cfg.Traces.PeerTags
 	acfg.MaxSenderRetries = 4
+	if traceCustomHTTPFeatureGate.IsEnabled() {
+		params.Logger.Warn("Experimental feature: datadog exporter trace export uses a custom HTTP client from the exporter HTTP configs")
```
Comment (mx-psi) [inline exporter/datadogexporter/traces_exporter.go:224]: This warning is not actionable, some people want to have a way to remove warnings if they are not actionable. Would it be fine to make it info level?
```suggestion
		params.Logger.Info("Experimental feature: datadog exporter trace export uses a custom HTTP client from the exporter HTTP configs")
```

File: exporter/datadogexporter/traces_exporter.go (hunk 1 above)
Comment (songy23) [reply exporter/datadogexporter/traces_exporter.go]: Renamed to `TraceExport...`

**Review by mx-psi — APPROVED** [review]
